app-modules/group/src/Policies/GroupPolicy.php

```diff
@@ -60,83 +60,83 @@ public function viewAny(Authenticatable $authenticatable): Response
     {
         return $authenticatable->canOrElse(
             abilities: 'segment.view-any',
-            denyResponse: 'You do not have permission to view segments.'
+            denyResponse: 'You do not have permission to view groups.'
         );
     }
 
-    public function view(Authenticatable $authenticatable, Segment $segment): Response
+    public function view(Authenticatable $authenticatable, Group $group): Response
     {
-        if (! $authenticatable->hasLicense($segment->model?->class()::getLicenseType())) {
-            return Response::deny('You do not have permission to view this segment.');
+        if (! $authenticatable->hasLicense($group->model->class()::getLicenseType())) {
+            return Response::deny('You do not have permission to view this group.');
         }
 
         return $authenticatable->canOrElse(
-            abilities: ["segment.{$segment->getKey()}.view"],
-            denyResponse: 'You do not have permission to view this segment.'
+            abilities: ["segment.{$group->getKey()}.view"],
```

Comment:
Since we are in this file now, please change this to the `*` version of the permission check, don't pass the key